tooling/SecretManager.VisualStudio/Sources/JsonConfigurationFileParser.cs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.SecretManager;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Extensions.Configuration.Json
+{
+    internal class JsonConfigurationFileParser
+    {
+        private readonly IDictionary<string, string> _data = new SortedDictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        private readonly Stack<string> _context = new Stack<string>();
+        private string _currentPath;
+
+        private JsonTextReader _reader;
+
+        public IDictionary<string, string> Parse(Stream input)
```

Comment:
This method should be static

Reply:
Can't be. The parser has context that is unique per instance.

Reply:
Yeah, this method should new up the parser 👍

Reply:
It's worth pointing out this is an exact copy of https://github.com/aspnet/Configuration/blob/dev/src/Config.Json/JsonConfigurationProvider.cs.

I'm fine with the change, but want to avoid making changes here unless we plan to port back to aspnet/Configuration. Can we do that change in aspnet/Configuration first and then port it here?

Reply:
Updated to be static. I'll add this change to https://github.com/aspnet/Configuration/pull/750 as well